xarray/core/dataset.py

```diff
@@ -5396,5 +5402,105 @@ def filter_by_attrs(self, **kwargs):
                 selection.append(var_name)
         return self[selection]
 
+    def unify_chunks(self) -> "Dataset":
+        """ Unifies chunksize along all chunked dimensions of this Dataset.
+
+        Returns
+        -------
+
+        Dataset with consistent chunk sizes for all dask-array variables
+
+        See Also
+        --------
+
+        dask.array.core.unify_chunks
+        """
+
+        import dask
+
+        if not dask.is_dask_collection(self):
+            return self.copy()
+
+        try:
+            self.chunks
+            return self.copy()
+        except ValueError:  # "inconsistent chunks"
+            pass
+
+        import dask.array
```

Comment:
can we move this import to the top of the method?

Reply:
@crusaderky wanted it here to save time I think.

Reply:
No, I wanted to allow using the function when dask is not installed and there is no dask backend.
But that's nullified by 
```python

        import dask

        if not dask.is_dask_collection(self):
            return self.copy()
```
right above (it was not there when I made the comment)!

Don't we have a wrapper around ``dask.is_dask_collection`` that defaults to a dummy False if dask is not installed?

Reply:
Correction - that whole paragraph can be flat out removed.
Dataset.chunks returns an empty ``Frozen(SortedKeysDict({}))`` when there are no dask variables already.

Reply:
🤦‍♂ Now changed to

``` python
        try:
            # if pure numpy
            if len(self.chunks) == 0:
                return self.copy()
        except ValueError:  # "inconsistent chunks"
            pass
```

Reply:
Wrong again - if you have chunks and they are already consistent, the self.chunks property will return a non-empty mapping.

```python
        try:
            self.chunks
        except ValueError:  # "inconsistent chunks"
            pass
        else:
            # No variables with dask backend, or all chunks are already aligned
            return self.copy()

        # import dask is placed after the quick exit test above to allow
        # running this method if dask isn't installed and there are no chunks
        import dask.array
```

Reply:
ugh, forgot about that.